operator/internal/manifests/storage/options.go

```diff
@@ -1,12 +1,28 @@
 package storage
 
+// SharedStoreOptions is used to configure
+// the boltdb_shipper store
+type SharedStoreOptions string
+
+var (
+	// SharedStoreAzure storage
+	SharedStoreAzure SharedStoreOptions = "azure"
+	// SharedStoreGCS storage
+	SharedStoreGCS SharedStoreOptions = "gcs"
+	// SharedStoreS3 storage
+	SharedStoreS3 SharedStoreOptions = "s3"
+	// SharedStoreSwift storage
+	SharedStoreSwift SharedStoreOptions = "swift"
+)
```

Comment:
Why do we need to replicate the same types here, as per API defs deliver the enum already? You could pass the type as an additional field into each `StorageConfig` WDYT?
```yaml
// ObjectStorageSecretType defines the type of storage which can be used with the Loki cluster.
//
// +kubebuilder:validation:Enum=azure;gcs;s3;swift
type ObjectStorageSecretType string

const (
	// ObjectStorageSecretAzure when using Azure for Loki storage
	ObjectStorageSecretAzure ObjectStorageSecretType = "azure"

	// ObjectStorageSecretGCS when using GCS for Loki storage
	ObjectStorageSecretGCS ObjectStorageSecretType = "gcs"

	// ObjectStorageSecretS3 when using S3 for Loki storage
	ObjectStorageSecretS3 ObjectStorageSecretType = "s3"

	// ObjectStorageSecretSwift when using Swift for Loki storage
	ObjectStorageSecretSwift ObjectStorageSecretType = "swift"
)
```